### PR TITLE
Fix peer score not being calculated when adding outbound connections

### LINKED ISSUE #364: Peer score is not calculated when a new outbound connection is created
If a connection is created using `GetConnection` and not used immediately (e.g. it's not created in response to a call or ping, but because the user explicitly created a connection), we weren't recalculating the peer's score.

This only affects outbound connections, inbound connections have their score calculated correctly as soon as the connection is created.

cc @yurishkuro

### PR DESCRIPTION
**Pull request by prashantv**
Use the same connectionEvents.onActive path for adding outbound
connections to a peer, and update the score. This makes the
inbound and outbound paths similar.

Fixes #364

cc @yurishkuro

### CODE REVIEW

**Comment by yurishkuro** [comment]
lgtm

File: channel.go [hunk 1]
```diff
@@ -540,20 +541,35 @@ func (ch *Channel) updatePeer(p *Peer) {
...
+func (ch *Channel) outboundConnectionActive(c *Connection) {
+	c.log.Debugf("Add outbound connection as an active peer for %v", c.remotePeerInfo.HostPort)
+	p := ch.rootPeers().GetOrAdd(c.remotePeerInfo.HostPort)
+	if err := p.AddOutboundConnection(c); err != nil {
+		c.log.WithFields(LogField{"remoteHostPort", c.remotePeerInfo.HostPort}).
+			Warn("Failed to add outbound connection to peer")
+	}
+
+	ch.updatePeer(p)
+}
```
Comment (akshayjshah) [inline channel.go]: Where are we adding the connection to the channel's mutable conns map?

Comment (prashantv) [inline channel.go]: Refactored things a little.

**Comment by prashantv** [comment]
Updated

**Comment by akshayjshah** [comment]
:shipit: Love it.

**Pull request merged**